auth0/resource_auth0_action.go

```diff
@@ -251,6 +259,61 @@ func deployAction(ctx context.Context, d *schema.ResourceData, m interface{}) di
 	return diag.FromErr(d.Set("version_id", actionVersion.GetID()))
 }
 
+func preventErasingUnmanagedSecrets(d *schema.ResourceData, api *management.Management) diag.Diagnostics {
+	if !d.HasChange("secrets") {
+		return nil
+	}
+
+	preUpdateAction, err := api.Action.Read(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oldSecrets, newSecrets := d.GetChange("secrets")
+
+	return checkForUnmanagedActionSecrets(
+		oldSecrets.([]interface{}),
+		newSecrets.([]interface{}),
```

Comment:
What is the reasoning behind casting these secrets to `[]interface{}`? Especially since these arguments get casted into a `map[string]interface{}` below. As an actionable takeaway, I think I'd prefer to see the type casting consolidated into a singular step and/or the parameter types made to be more explicit. Otherwise, it's difficult to follow the flow of logic.

Reply:
> What is the reasoning behind casting these secrets to []interface{}?

The secrets are of type `schema.TypeList`, and that converts to a `[]interface{}` within Go. 

> Especially since these arguments get casted into a map[string]interface{} below. 

They do not, it's their elements that get casted to a `map[string]interface{}`. 

> As an actionable takeaway, I think I'd prefer to see the type casting consolidated into a singular step and/or the parameter types made to be more explicit. Otherwise, it's difficult to follow the flow of logic.

We can't, because the secrets look like this actually:

```go
secrets := []interface{}{
  map[string]interface{}{
    "name": "secretName",
    "value": "secretValue",
  },
} 
// secrets is a []interface{} with len 1
// it has an element of type map[string]interface{}
```
So first we need to typecast the most outer object and then again typecast each element of the slice.